DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/README.md

```diff
@@ -0,0 +1,358 @@
+
+Use these commands to run the design, depending on your OS.
+
+### On a Linux* System
+This design uses CMake to generate a build script for GNU/make.
+
+1. Change to the sample directory.
+
+2. For different parts of this tutorial, navigate to the appropriate sub-folder.
+   ```
+   cd <partX_XXX>
+   ```
+   where `<partX_XXX>` is:
+   - `part1_pointers`
+   - `part2_single_host`
+   - `part3_hosts`
+   - `part4_ddr_hosts`
+
+3. Configure the build system for the Agilex® 7 device family, which is the default.
+
+   ```
+   mkdir build
+   cd build
+   cmake ..
+   ```
+
+   > **Note**: You can change the default target by using the command:
+   >  ```
+   >  cmake .. -DFPGA_DEVICE=<FPGA device family or FPGA part number>
+   >  ```
+
+4. Compile the design using `make`.
+   | Compilation Type    | Command
+   |:---                 |:---
+   | FPGA Emulator       | `make fpga_emu`
+   | Optimization Report | `make report`
+   | FPGA Simulator      | `make fpga_sim`
+   | FPGA Hardware       | `make fpga`
+
+### On a Windows* System
+This design uses CMake to generate a build script for  `nmake`.
+
+1. Change to the sample directory.
+2. For different parts of this tutorial, navigate to the appropriate sub-folder.
+   ```
+   cd <partX_XXX>
+   ```
+   where `<partX_XXX>` is:
+   - `part1_pointers`
+   - `part2_single_host`
+   - `part3_hosts`
+   - `part4_ddr_hosts`
+
+3. Configure the build system for the Agilex® 7 device family, which is the default.
+   ```
+   mkdir build
+   cd build
+   cmake -G "NMake Makefiles" ..
+   ```
+   > **Note**: You can change the default target by using the command:
+   >  ```
+   >  cmake -G "NMake Makefiles" .. -DFPGA_DEVICE=<FPGA device family or FPGA part number>
+   >  ```
+3. Compile the design using `nmake`.
+   | Compilation Type    | Command (Windows)
+   |:---                 |:---
+   | FPGA Emulator       | `nmake fpga_emu`
+   | Optimization Report | `nmake report`
+   | FPGA Simulator      | `nmake fpga_sim`
+   | FPGA Hardware       | `nmake fpga`
+   > **Note**: If you encounter any issues with long paths when compiling under Windows*, you may have to create your ‘build’ directory in a shorter path, for example c:\samples\build.  You can then run cmake from that directory, and provide cmake with the full path to your sample directory.
+
+## Examining the Reports
+Locate `report.html` in the `<source_file>.prj/reports/` directory. Open the report in Chrome*, Firefox*, Edge*, or Internet Explorer*.
+
+Navigate to the Area Analysis section of the optimization report. The Kernel System section displays the area consumption of each kernel. Notice that the `MultiMMIP` kernel consumes way less area under all categories than the `PointerIP` kernel. This is due to stall free memory accesses and the removal of arbitration logic, both of which come from separating the accesses into their own interfaces.
```

Comment:
```suggestion
Navigate to the Area Analysis section of the optimization reports for `part1_pointers` and `part3_hosts`. The Kernel System section displays the area consumption of each kernel. Notice that the `MultiMMIP` kernel consumes less area under all categories than the `PointerIP` kernel. This is due to stall-free memory accesses and the removal of arbitration logic. The fixed-latency on-chip block RAMs can be accessed with stall-free load/store units (LSUs), and giving each memory access a single dedicated interface allows the removal of arbitration logic.
```

stall-free memory accesses should be due to choosing **fixed-latency** memories, not just from separate interfaces. Variable-latency Avalon memory-mapped host interfaces should not be stall-free. Please open a bug if you see that variable-latency memory-mapped host interfaces are being classified as stall-free.